flow/set.go

```diff
@@ -252,20 +247,47 @@ func (fs *FlowSet) Dedup(field string) error {
 		return nil
 	}
 
-	visited := make(map[interface{}]bool)
+	dedupFlows := NewDedup(&deduped, field)
+	if err := dedupFlows.Flows(fs.Flows); err != nil {
+		return err
+	}
+	fs.Flows = deduped
+
+	return nil
+}
+
+// Group return only the flows that have a matching field in a FlowSet
+func (fs *FlowSet) Group(field string) error {
+	group := make(map[string][]*Flow, len(fs.Flows))
+
+	if len(fs.Flows) == 0 {
+		return nil
+	}
+	if field == "" {
+		return nil
+	}
+
+	var sortedFields []string
 	for _, flow := range fs.Flows {
-		kvisited, err := getDedupField(flow, field)
+		fieldValue, err := flow.GetFieldString(field)
 		if err != nil {
-			return err
+			/* ignore flow that have the field not set, like ARP packet don't have Network set */
+			continue
 		}
-
-		if _, ok := visited[kvisited]; !ok {
-			deduped = append(deduped, flow)
-			visited[kvisited] = true
+		if len(group[fieldValue]) == 0 {
+			sortedFields = append(sortedFields, fieldValue)
 		}
+		group[fieldValue] = append(group[fieldValue], flow)
 	}
-	fs.Flows = deduped
 
+	var groupFlows []*Flow
+	for _, field := range sortedFields {
+		flows := group[field]
+		if len(flows) > 1 {
```

Comment:
Why > 1 ? I don't understand the purpose of this, can you elaborate on why you keep only flow having the "groupby" seen at least twice ?

Reply:
see above